### ISSUE TITLE
Support DeepLab ONNX model as exported by PyTorch's dynamo-based exporter

### ISSUE THREAD

**Issue by robertknight** [robertknight/rten#645]
Changing the [DeepLab ONNX export script](https://github.com/robertknight/rten/blob/ee70cecf924df6c3774db5e38c661860ba0924da/rten-examples/src/export-deeplab.py#L20) to use the new dynamo-based exporter, by adding `dynamo=True` to the `torch.export` call, produces an ONNX model which `rten-convert` cannot convert. This issue tracks the various features which are missing to make it work.

- [x] Support for `Constant` operators which store their value in an attribute other than `value` (eg. `value_ints`, `value_floats`). See [spec](https://onnx.ai/onnx/operators/onnx__Constant.html) (https://github.com/robertknight/rten/pull/649)
- [x] Support `CastLike` operator (https://github.com/robertknight/rten/pull/646)
- [x] Support for  `start` and `end` attributes in `Shape` operator (https://github.com/robertknight/rten/pull/648)
- [x] Support for `pytorch_half_pixel` value for `coordinate_transformation_mode` attribute of `Resize` operator (https://github.com/robertknight/rten/pull/647)
- [ ] Support for function operators (eg. `_aten_native_batch_norm_inference_onnx` in this model). _Update: This requirement goes away if the model is optimized after export. See https://github.com/robertknight/rten/pull/651. After fixing that a subsequent issue arises where inference produces different results than for TorchScript._
- [x] Investigate different inference result when comparing optimized Dynamo-exported model against TorchScript-exported model (https://github.com/robertknight/rten/pull/654)

**Comment by robertknight** [comment]
I picked DeepLab for testing Dynamo-exported models because there is a convenient script in `rten-examples/src/export_deeplab.py` for exporting. Exporting other models with the Dynamo exporter will likely encounter additional issues.

**Comment by robertknight** [comment]
From testing a few timm models exported with `tools/export-timm-model.py` the most common issues with other vision models are:

- Lack of support for functions
- Lack of support for `Dropout` operator. This is often a no-op node since the dropout ratio is zero. Optimizing the ONNX model should remove this operator, but it makes sense to support it anyway.

**Comment by robertknight** [comment]
As of https://github.com/robertknight/rten/pull/654 it is possible to export the DeepLab v3 model using the TorchDynamo exporter, convert it and then run it.
